src/utils/getDataFromServer.js

```diff
@@ -0,0 +1,31 @@
+import {categoryUrl,LOAD_CATEGORY,IS_LOADING,ERROR,productUrl,LOAD_PRODUCTS} from "../utils/constants";
+import axios from "axios";
+
+export const getCategoryFromServer=async(dispatch)=>{
+ 
+    try {
+        const response = await axios.get(categoryUrl);
+        if (response.status === 200) {
+            console.log(response.data.categories);
+            dispatch({type:LOAD_CATEGORY,payload:response.data.categories});
+        }else{
+            throw new Error("Failed to load category");
```

Comment:
Instead of throwing an error, you can utilize this error for showing the Not found page on User UI.

Reply:
This part i will implement later on for now removed this part of code